core/src/main/java/org/wso2/msf4j/util/Utils.java

```diff
@@ -114,4 +114,38 @@ public static int getIntersection(Set set1, Set set2) {
         cloneSet2.retainAll(cloneSet1);
         return cloneSet2.size();
     }
+
+    /**
+     * Get the path without leading or trailing / and with multiple / reduced to one /
+     *
+     * @param path The path
+     * @return normalized path
+     */
+    public static final String normalizePath(String path) {
```

Comment:
The logic here is too complex IMO. What do you think if we can simplify this code as below?

```java
    public static String normalizePath(String path) {

        // Replace multiple '/' characters with a single '/'
        path = path.replaceAll("[//]+", "/");

        // Remove leading '/'
        if (path.startsWith("/")) {
            path = path.substring(1, path.length());
        }

        // Remove trailing '/'
        if (path.endsWith("/")) {
            path = path.substring(0, path.length() - 1);
        }

        return path;
    }
```

Reply:
Your `"[//]+"` matches the same as `"/+"`, did you mean `"//+"`? All comes out to the same behavior though.

We could simplify that even further to
```
return path.replaceAll("/+", "/").replaceAll("^/", "").replaceAll("/$", "");
```
or (a bit more cryptic, but one fewer call)
```
return path.replaceAll("/+", "/").replaceAll("^/|/$", "");
```
Slightly less efficient than the original code, but much more readable.  I would be happy to change it to that.